diff --git a/CMakeLists.txt b/CMakeLists.txt
index ba4078d..c1dcd85 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -140,9 +140,9 @@ elseif (APPLE AND CMAKE_SYSTEM_PROCESSOR STREQUAL "arm64")
 else()
   if (SKBUILD)
     # Reasonably portable binaries for PyPI
-    set(${P}_NATIVE_FLAGS_DEFAULT "-march=ivybridge")
+    # set(${P}_NATIVE_FLAGS_DEFAULT "-march=ivybridge")
   else()
-    set(${P}_NATIVE_FLAGS_DEFAULT "-march=native")
+    # set(${P}_NATIVE_FLAGS_DEFAULT "-march=native")
   endif()
 endif()
 