From c0845cd8e7541bd15347459fa8d97eaf55b0f49c Mon Sep 17 00:00:00 2001
From: Jeongseok Lee <jeongseok@meta.com>
Date: Fri, 19 Sep 2025 02:29:22 -0700
Subject: [PATCH 3/4] Make directory before creating files

---
 src/python/CMakeLists.txt | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/python/CMakeLists.txt b/src/python/CMakeLists.txt
index 52a13082..eb7240e4 100644
--- a/src/python/CMakeLists.txt
+++ b/src/python/CMakeLists.txt
@@ -255,6 +255,7 @@ if (NOT (DRJIT_SANITIZE_ASAN OR DRJIT_SANITIZE_UBSAN))
       ${DRJIT_PYTHON_DST_DIR}/auto/__init__.py
       ${DRJIT_PYTHON_DST_DIR}/auto/ad.py
 
+    COMMAND ${CMAKE_COMMAND} -E make_directory ${DRJIT_PYTHON_DST_DIR}/auto
     COMMAND cmake -E touch
       ${DRJIT_PYTHON_DST_DIR}/auto/py.typed
       ${DRJIT_PYTHON_DST_DIR}/auto/__init__.py
-- 
2.51.0

