diff --git a/CMakeLists.txt b/CMakeLists.txt
index 47d0f6b..29fe9e5 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -137,11 +137,11 @@ else()
   if (SKBUILD)
     if (CMAKE_SYSTEM_PROCESSOR MATCHES "aarch64|arm64|ARM64")
-      set(${P}_NATIVE_FLAGS_DEFAULT "-march=armv8-a")
+      # set(${P}_NATIVE_FLAGS_DEFAULT "-march=armv8-a")
     else()
-      set(${P}_NATIVE_FLAGS_DEFAULT "-march=haswell")
+      # set(${P}_NATIVE_FLAGS_DEFAULT "-march=haswell")
     endif()
   else()
-    set(${P}_NATIVE_FLAGS_DEFAULT "-march=native")
+    # set(${P}_NATIVE_FLAGS_DEFAULT "-march=native")
   endif()
 endif()
 
