From 3506fed9b4b4709ac33527513b23b037de0cfe17 Mon Sep 17 00:00:00 2001
From: Jeongseok Lee <jeongseok@meta.com>
Date: Mon, 5 May 2025 16:59:25 -0700
Subject: [PATCH 2/2] Fix install destination paths

---
 CMakeLists.txt | 26 +++++++++++++++++++++++---
 1 file changed, 23 insertions(+), 3 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 6288e844..38bc51aa 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -195,9 +195,29 @@ if (DRJIT_MASTER_PROJECT)
   install(DIRECTORY include/drjit DESTINATION ${CMAKE_INSTALL_INCLUDEDIR})
   install(DIRECTORY ext/drjit-core/include/drjit-core DESTINATION ${CMAKE_INSTALL_INCLUDEDIR})
 
-  install(TARGETS drjit-core EXPORT drjitTargets DESTINATION drjit)
-  install(TARGETS drjit-extra EXPORT drjitTargets DESTINATION drjit)
-  install(TARGETS nanothread EXPORT drjitTargets DESTINATION drjit)
+  if (DRJIT_ENABLE_JIT)
+    install(
+      TARGETS drjit-core
+      EXPORT drjitTargets
+      RUNTIME DESTINATION "${CMAKE_INSTALL_BINDIR}"
+      ARCHIVE DESTINATION "${CMAKE_INSTALL_LIBDIR}"
+      LIBRARY DESTINATION "${CMAKE_INSTALL_LIBDIR}"
+    )
+    install(
+      TARGETS drjit-extra
+      EXPORT drjitTargets
+      RUNTIME DESTINATION "${CMAKE_INSTALL_BINDIR}"
+      ARCHIVE DESTINATION "${CMAKE_INSTALL_LIBDIR}"
+      LIBRARY DESTINATION "${CMAKE_INSTALL_LIBDIR}"
+    )
+    install(
+      TARGETS nanothread
+      EXPORT drjitTargets
+      RUNTIME DESTINATION "${CMAKE_INSTALL_BINDIR}"
+      ARCHIVE DESTINATION "${CMAKE_INSTALL_LIBDIR}"
+      LIBRARY DESTINATION "${CMAKE_INSTALL_LIBDIR}"
+    )
+  endif()
 
   if (DRJIT_ENABLE_PYTHON)
     install(TARGETS drjit-python DESTINATION drjit)
-- 
2.49.0.windows.1

